pyqtorch/utils.py

```diff
@@ -142,3 +143,31 @@ def density_mat(state: Tensor) -> Tensor:
     state = torch.permute(state, batch_first_perm).reshape(batch_size, 2**n_qubits)
     undo_perm = (1, 2, 0)
     return torch.permute(torch.einsum("bi,bj->bij", (state, state.conj())), undo_perm)
+
+
+def promote_ope(operator: Tensor, target: int, n_qubits: int) -> Tensor:
+    """
+    Promotes `operator` to the size of the circuit.
+
+    Args:
+        operator (Tensor): The operator tensor to be promoted.
+        target (int): The target qubit index.
+        n_qubits (int): Number of qubits in the circuit.
+
+    Returns:
+        Tensor: The promoted operator tensor.
+
+    Raises:
+        ValueError: If `target` is not within the range of qubits.
+    """
+    if target > n_qubits:
+        raise ValueError("The target must be a register qubit")
+
+    qubits_support = torch.arange(0, n_qubits)
```

Comment:
You can maybe pass the whole `qubit_support` as opposed to deducing it from `n_qubits` ?

Reply:
`qubit_support` is defined in `Primitive` as : `qubit_support = (target,)` and not as the list of all the qubit using in the system.
So maybe changing the name variable as: `qubits` is more appropriate right ?

Reply:
Done, check #155.